### PR TITLE
Use migrations manifest

### PR DESCRIPTION
**Pull request by pjdufour-truss**
## Description

Rather than running whatever files are in the migrations folders, `milmove migrate` will now error on any files that were not listed in the migrations manifest.

Added a pre-commit hook, that updates the manifest.  To avoid issues with merge commits, the new pre-commit hook doesn't run in CI.  However, if a commit is missing from the manifest the server tests will fail.

We'll have the added benefit of an easy to view changelog of migrations.

## Reviewer Notes

**Local Migrations**

```shell
make db_dev_destroy && make db_dev_start && DB_TIMEOUT=30 make db_dev_create && make db_dev_migrate
```

## Setup

None

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/tree/master/docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested using `scripts/run-prod-migrations`
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

None

## Screenshots

None

### CODE REVIEW

**Comment by pjdufour-truss** [comment]
Hard to see how to add unit tests..., since fizz automatically executes migrations as they are parsed.

**Comment by pjdufour-truss** [comment]
Now requires https://github.com/gobuffalo/pop/pull/407

File: go.mod [hunk 1]
```diff
@@ -73,13 +78,15 @@ require (
...
 	golang.org/x/text v0.3.2
-	golang.org/x/tools v0.0.0-20190511041617-99f201b6807e
+	golang.org/x/tools v0.0.0-20190621195816-6e04913cbbac
 	gopkg.in/alexcesaro/quotedprintable.v3 v3.0.0-20150716171945-2caba252f4dc // indirect
 	gopkg.in/alexcesaro/statsd.v2 v2.0.0 // indirect
 	gopkg.in/gomail.v2 v2.0.0-20160411212932-81ebce5c23df // indirect
 	gopkg.in/urfave/cli.v1 v1.20.0 // indirect
 )
+
+replace github.com/gobuffalo/pop => github.com/pjdufour-truss/pop v4.11.2-0.20190611171623-8b1bc4b6292d+incompatible
```
Comment (rdhariwal) [inline go.mod]: Is this a place holder until this [PR gobuffalo/pop#407](https://github.com/gobuffalo/pop/pull/407) merges?

Comment (pjdufour-truss) [reply go.mod]: yes

**Comment by chrisgilmerproj** [comment]
@jim - would just modifying this script help? https://github.com/transcom/mymove/blob/master/scripts/gen-model . I'm thinking `scripts/gen-migration` might be as simple as `touch migrations/$(date '+%Y%m%d')_new_migration_name.up.fizz && scripts/update-migrations-manifest`

**Comment by jim** [comment]
@chrisgilmerproj I don't think we generate models that much anymore, but we do create lots of other migrations. We could probably replace `gen-model` with `gem-migration` if we wanted.

File: pkg/cli/migration_file.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+func (e *errInvalidMigrationType) Error() string {
+	return fmt.Sprintf("invalid migration type %q, expecting sql or fizz.", e.Value)
+}
+
+// InitMigrationFlags initializes the Migration command line flags
+func InitMigrationFileFlags(flag *pflag.FlagSet) {
+	flag.String(MigrationVersionFlag, time.Now().Format(VersionTimeFormat), "migration version: integer representation of datetime, default is current time using Go format "+VersionTimeFormat)
+	flag.StringP(MigrationNameFlag, "n", "", "migration name: alphanumeric, no spaces, underscores and dashes allowed")
+	flag.StringP(MigrationTypeFlag, "t", "", "migration type: fizz or sql.")
```
Comment (chrisgilmerproj) [inline pkg/cli/migration_file.go]: I think I'd prefer that this had a default and that it was `fizz` which is what most people use.

File: pkg/cli/migration_file.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+const (
+	// MigrationVersionFlag is the migration version flag
+	MigrationVersionFlag string = "version"
+	// MigrationNameFlag is the migration path flag
+	MigrationNameFlag string = "name"
+	// MigrationTypeFlag is the migration manifest flag
+	MigrationTypeFlag string = "type"
+
+	VersionTimeFormat string = "20060102150405"
```
Comment (chrisgilmerproj) [inline pkg/cli/migration_file.go]: Needs a comment here :)

File: cmd/milmove/main.go [hunk 4]
```diff
@@ -50,11 +50,31 @@ func main() {
...
 	initMigrateFlags(migrateCommand.Flags())
 	root.AddCommand(migrateCommand)
 
+	genCommand := &cobra.Command{
+		Use:   "gen",
+		Short: "Generate migrations and other objects",
+		Long:  "Generate migrations and other objects",
+		RunE:  nil,
+	}
+	initMigrateFlags(genCommand.Flags())
```
Comment (chrisgilmerproj) [inline cmd/milmove/main.go]: Do we want `milmove gen -h` to actually have any options?

File: pkg/cli/migration_file.go (hunk 2 above)
Comment (pjdufour-truss) [reply pkg/cli/migration_file.go]: DOne

File: pkg/cli/migration_file.go (hunk 3 above)
Comment (pjdufour-truss) [reply pkg/cli/migration_file.go]: Done

File: cmd/milmove/main.go (hunk 4 above)
Comment (pjdufour-truss) [reply cmd/milmove/main.go]: Fixed

**Comment by jim** [comment]
@pjdufour-truss I was expecting a shell script, so this is great!

**Pull request merged**